src/extension.ts

```diff
@@ -34,31 +34,51 @@ export function activate(context: vscode.ExtensionContext) {
 
     context.subscriptions.push(disposable1);
 
-    const disposable2 = vscode.commands.registerCommand('ucl-rsdg.validateAtf', async () => {
-        const editor = vscode.window.activeTextEditor;
-        const fileName = basename(editor.document.uri.fsPath);
-        const fileContent = editor.document.getText();
-        let fileProject: string;
-        try {
-            fileProject = getProjectCode(fileContent);
-        } catch (err) {
-            vscode.window.showErrorMessage(`Could not validate: ${err}`);
-            return;
-        }
-        // The validate function is currently not mapped to the appropriate logging functions
-        const result = await validate(fileName,fileProject,fileContent);
-        handleResult(result, editor);
+    const disposable2 = vscode.commands.registerCommand('ucl-rsdg.validateAtf', () => {
+        workWithServer("validate", validate);
         });
 
     context.subscriptions.push(disposable2);
 
+    context.subscriptions.push(
+        vscode.commands.registerCommand('ucl-rsdg.lemmatiseAtf', () => {
+            workWithServer("lemmatise", lemmatise)
```

Comment:
`send_command` instead of `workWithServer` maybe?

Reply:
Yes, maybe that's better. Although this doesn't only send the command, it does some things before and after too.